File: app/src/main/java/torille/fi/lurkforreddit/comments/CustomLinkTransformationMethod.java
package torille.fi.lurkforreddit.comments;

import android.graphics.Rect;
import android.text.Spannable;
import android.text.method.TransformationMethod;
import android.text.style.URLSpan;
import android.view.View;
import android.widget.TextView;

/**
 * Transforms links with the help of {@link CustomUrlSpan}
 */

class CustomLinkTransformationMethod implements TransformationMethod {
    @Override
    public CharSequence getTransformation(CharSequence source, View view) {
        if (view instanceof TextView) {
            TextView textView = (TextView) view;
            //LinkifyCompat.addLinks(textView, Linkify.WEB_URLS);
            if (textView.getText() == null || !(textView.getText() instanceof Spannable)) {
                return source;
            }
            Spannable text = (Spannable) textView.getText();
            URLSpan[] spans = text.getSpans(0, textView.length(), URLSpan.class);
            for (int i = spans.length - 1; i >= 0; i--) {
                URLSpan oldSpan = spans[i];
                int start = text.getSpanStart(oldSpan);
                int end = text.getSpanEnd(oldSpan);
                int flags = text.getSpanFlags(oldSpan);
                String url = oldSpan.getURL();
                text.removeSpan(oldSpan);
                text.setSpan(new CustomUrlSpan(url), start, end, flags);
            }
            return text;
        }
        return source;
    }

    @Override
    public void onFocusChanged(View view, CharSequence sourceText, boolean focused, int direction, Rect previouslyFocusedRect) {

    }
}

File: app/src/main/java/torille/fi/lurkforreddit/comments/CustomLinkTransformationMethod.kt
package torille.fi.lurkforreddit.comments

import android.graphics.Rect
import android.text.Spannable
import android.text.method.TransformationMethod
import android.text.style.QuoteSpan
import android.text.style.URLSpan
import android.view.View
import android.widget.TextView

/**
 * Transforms links with the help of [CustomUrlSpan]
 */

internal class CustomLinkTransformationMethod(val color: Int) : TransformationMethod {
    override fun getTransformation(source: CharSequence, view: View): CharSequence {
        if (view is TextView) {
            if (view.text == null || view.text !is Spannable) {
                return source
            }
            val text = view.text as Spannable
            val spans = text.getSpans(0, view.length(), URLSpan::class.java)
            for (i in spans.indices.reversed()) {
                val oldSpan = spans[i]
                val start = text.getSpanStart(oldSpan)
                val end = text.getSpanEnd(oldSpan)
                val flags = text.getSpanFlags(oldSpan)
                val url = oldSpan.url
                text.removeSpan(oldSpan)
                text.setSpan(CustomUrlSpan(url, color), start, end, flags)
            }
            val quoteSpan = text.getSpans(0, view.length(), QuoteSpan::class.java)
            for (quote in quoteSpan.indices.reversed()) {
                val oldSpan = quoteSpan[quote]
                val start = text.getSpanStart(oldSpan)
                val end = text.getSpanEnd(oldSpan)
                val flags = text.getSpanFlags(oldSpan)
                text.removeSpan(oldSpan)
                text.setSpan(QuoteSpan(color), start, end, flags)
            }
            return text
        }
        return source
    }

    override fun onFocusChanged(
        view: View,
        sourceText: CharSequence,
        focused: Boolean,
        direction: Int,
        previouslyFocusedRect: Rect
    ) {
    }
}
